presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftDomain.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.presto.spi.predicate.AllOrNoneValueSet;
+import com.facebook.presto.spi.predicate.Domain;
+import com.facebook.presto.spi.predicate.EquatableValueSet;
+import com.facebook.presto.spi.predicate.SortedRangeSet;
+import com.facebook.presto.spi.predicate.ValueSet;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import static com.facebook.presto.thrift.interfaces.client.ThriftAllOrNoneValueSet.fromAllOrNoneValueSet;
+import static com.facebook.presto.thrift.interfaces.client.ThriftAllOrNoneValueSet.toAllOrNoneValueSet;
+import static com.facebook.presto.thrift.interfaces.client.ThriftEquatableValueSet.fromEquatableValueSet;
+import static com.facebook.presto.thrift.interfaces.client.ThriftEquatableValueSet.toEquatableValueSet;
+import static com.facebook.presto.thrift.interfaces.client.ThriftRangeValueSet.fromSortedRangeSet;
+import static com.facebook.presto.thrift.interfaces.client.ThriftRangeValueSet.toRangeValueSet;
+import static com.facebook.swift.codec.ThriftField.Requiredness.OPTIONAL;
+import static com.google.common.base.Preconditions.checkArgument;
+
+@ThriftStruct
+public final class ThriftDomain
+{
+    private final ThriftAllOrNoneValueSet allOrNoneValueSet;
+    private final ThriftEquatableValueSet equatableValueSet;
+    private final ThriftRangeValueSet rangeValueSet;
+    private final boolean nullAllowed;
+
+    @ThriftConstructor
+    public ThriftDomain(
+            @Nullable @ThriftField(name = "allOrNoneValueSet") ThriftAllOrNoneValueSet allOrNoneValueSet,
```

Comment:
I had a weird NPE from swift when I had this and similar structures without names when static conversion methods were defined. Since then I moved some of static methods, so I can double check if it works now without explicit names.